### PR TITLE
Minor: make some physical-plan properties public

### LINKED ISSUE #12021: Make properties in phyiscal-plan/aggregates public
### Is your feature request related to a problem or challenge?

We're working on building out streaming functionality datafusion via the [Denormalized](https://github.com/probably-nothing-labs/denormalized) project. In order to support streaming aggregates, we need access to various properties and functions that are currently private, or public only to the physical-plan crate.

### Describe the solution you'd like

Make certain properties and functions public.

### Describe alternatives you've considered

Duplicating the relevant files into our code base so we can control the visibility of different methods, structs, and properties.

### Additional context

We're trying to avoid duplicating datafusion code into our codebase which often means we need to access things that are marked private.

### PR DESCRIPTION
**Pull request by emgeee**
## Which issue does this PR close?
Closes #12021

## Are there any user-facing changes?

No

### CODE REVIEW

File: datafusion/physical-plan/src/aggregates/mod.rs [hunk 1]
```diff
@@ -128,14 +128,14 @@ impl AggregateMode {
 #[derive(Clone, Debug, Default)]
 pub struct PhysicalGroupBy {
     /// Distinct (Physical Expr, Alias) in the grouping set
-    expr: Vec<(Arc<dyn PhysicalExpr>, String)>,
+    pub expr: Vec<(Arc<dyn PhysicalExpr>, String)>,
```
Comment (alamb) [inline datafusion/physical-plan/src/aggregates/mod.rs:131]: A potentially more future proof API might be to make accessors for these fields

like 
```rust
pub fn null_expr(&self) -> &[...]
```

File: datafusion/physical-plan/src/aggregates/order/full.rs [hunk 2]
```diff
@@ -142,3 +142,9 @@ impl GroupOrderingFull {
         std::mem::size_of::<Self>()
     }
 }
+
+impl Default for GroupOrderingFull {
```
Comment (alamb) [inline datafusion/physical-plan/src/aggregates/order/full.rs:146]: 👍

**Review by alamb — APPROVED** [review]
Makes sense to me -- thanks @emgeee

File: datafusion/physical-plan/src/aggregates/mod.rs (hunk 1 above)
Comment (emgeee) [reply datafusion/physical-plan/src/aggregates/mod.rs:131]: ah, i just noticed that these properties do have accessors already so let me revert this chagne

**Review by alamb — APPROVED** [review]
Thanks again @emgeee

**Pull request merged**